src/middlewared/middlewared/plugins/pool.py

```diff
@@ -755,6 +793,25 @@ def add(attr):
                 if i in data:
                     verrors.add(f'{schema}.{i}', 'This field is not valid for VOLUME')
 
+            if 'volsize' in data and parent:
```

Comment:
`parent` is always true, right?

Reply:
Nope, if mode is create then it's possible that it might not be true

Reply:
Care to elaborate? How can it ever not be true?

Reply:
Let me add, if you're creating a dataset, the main dataset will always exist, e.g. `tank`, you cannot create `tank/foo` if `tank` does not exist.